### PR TITLE
New `remove_objects_without_attrs` function

### PR DESCRIPTION
**Pull request by tylerganter**
Breaking this out from `filter_by_schema` and adding some modularity by specifying which labels to remove.

### CODE REVIEW

File: eta/core/datasets.py [hunk 1]
```diff
@@ -2751,20 +2751,21 @@ class SchemaFilter(DatasetTransformer):
     is None, no filtering is done.
     '''
 
-    def __init__(
-            self, schema, remove_objects_without_attrs=False,
-            prune_empty=True):
-        '''Initialize the SchemaFilter with a schema.
+    def __init__(self, schema, object_labels_to_filter=None, prune_empty=True):
```
Comment (brimoor) [inline eta/core/datasets.py]: With this refactor you'd have to set `object_labels_to_filter = []` if you wanted to NOT filter objects with no attributes. That's a bit confusing. I think you still need the `remove_objects_without_attrs` flag here to more intuitively control whether you want filtering to happen

Comment (tylerganter) [reply eta/core/datasets.py]: True, I added another boolean param to help clarify thtis

File: eta/core/image.py [hunk 2]
```diff
@@ -174,17 +174,25 @@ def merge_labels(self, image_labels):
...
+
+    def remove_objects_without_attrs(self, labels=None):
+        '''Removes DetectedObjects from this instance that do not have
+        attributes.
+
+        Args:
+            labels: an optional list of object labels to which to restrict
+                attention when filtering. By default, all objects are processed
+        '''
+        self.objects.remove_objects_without_attrs(labels=labels)
```
Comment (brimoor) [inline eta/core/image.py]: Note that I used the `labels=labels` syntax to invoke the `remove_objects_without_attrs` method.

When invoking methods with keyword arguments, please always use the `name=value` syntax. That way, if someone else comes along and adds a new keyword argument to the method that changes the order of the arguments, any existing invocations of that method will still work.

Technically, one should NOT insert keyword arguments anywhere besides the end of the list to guarantee backwards compatibility, but, hey, it happens. For example, you yourself added the new `object_labels_to_filter` argument to `SchemaFilter` BEFORE the existing `prune_empty` argument, which could break anyone's existing code that reads `SchemaFilter(schema, prune_empty)`

File: eta/core/datasets.py [hunk 3]
```diff
@@ -2751,20 +2751,25 @@ class SchemaFilter(DatasetTransformer):
...
         Args:
             schema: a VideoLabelsSchema or ImageLabelsSchema
             remove_objects_without_attrs: whether to remove objects with no
-                attributes, after filtering. By default, this is False
+                attributes, after filtering. Use the `object_labels_to_filter`
+                argument to control which object labels are filtered. By
+                default, this is False
+            object_labels_to_filter: an optional list of object labels for
+                which to restrict attention when filtering objects with no
+                attributes. If None, all objects are filtered
```
Comment (brimoor) [inline eta/core/datasets.py]: One other thing related to style- I see that you sometimes introduce docstrings that use the `parameter (type): description` syntax to document the type of the argument. I wish that all of ETA did that, as that would allow us to use Sphinx (as we do in https://github.com/voxel51/api-py) to generate auto-documentation that can parse out parameter types. However, we haven't adopted that in the existing codebase, so the style guide says to follow the existing docstring convention in the repo.

tl;dr: no `(type)` for now

File: eta/core/image.py [hunk 4]
```diff
@@ -184,16 +184,15 @@ def filter_by_schema(self, schema):
...
+        '''Removes DetectedObjects from this instance that do not have
+        attributes.
 
         Args:
-            object_labels_to_filter: a list of DetectedObject.label strings.
-                DetectedObject.labels not in this list are not removed even if
-                they have no attributes. If None, all object labels are
-                filtered.
+            labels: an optional list of object labels to which to restrict
+                attention when filtering. By default, all objects are processed
```
Comment (tylerganter) [inline eta/core/image.py]: I'm not keen on this description of the arg. I want to make it clear what each element of the list is, which is two things:
1) a string
2) a value that will be matched/compared against a `DetectedObject.label`
I'm indifferent to the exact syntax by which we convey that, but I think it makes this documentation a lot easier if these two snippets of information are conveyed.

Comment (brimoor) [reply eta/core/image.py]: do what you will. PR is approved for you
